### PR TITLE
Substitute for CheckReturnValue

### PR DESCRIPTION
**Pull request by dmdashenkov**
The `CheckReturnValue` ErrorProne check cannot be configured. Thus, when calling `builder.setValue(..)`, an error is produced, pointing out that the result of `setValue` should be used.

This is an issue for our message handler methods, which allow users to modify the entity builder. It is not supposed that users use the resulting value directly.

In order to tweak the checker, we extend it and add a custom condition. Now, invocations of methods starting with `set`, `add`, `put`, or `merge` on an instance of `Message.Builder` are ignored.

Also, this PR enables the `errorprone-checks` tests. For that, we publish artifacts into the maven local repository before the tests are run.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: base/build.gradle [hunk 1]
```diff
@@ -56,6 +56,13 @@ jar {
     from "$rootDir/base-validating-builders/compiled-proto"
 }
 
-build.doLast {
-    runBuild("$rootDir/base-validating-builders")
+task reBuildProtobuf {
```
Comment (armiol) [inline base/build.gradle]: I would go with `rebuild...` :)

File: tools/errorprone-checks/src/test/java/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+This tests require configuring `-Xbootclasspath...` option with the path to
```
Comment (armiol) [inline tools/errorprone-checks/src/test/java/README.md]: `These`

File: tools/errorprone-checks/src/test/java/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+This tests require configuring `-Xbootclasspath...` option with the path to 
+the `com.google.errorprone.javac` JAR.
+
+In Gradle this is done automatically via the separate task (see the `build.gradle` of this module).
+To run the test in Idea, you need to add the VM options manually in the "Edit configurations" tab.
```
Comment (armiol) [inline tools/errorprone-checks/src/test/java/README.md]: `IDEA`.

File: tools/errorprone-checks/src/test/java/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+This tests require configuring `-Xbootclasspath...` option with the path to 
+the `com.google.errorprone.javac` JAR.
+
+In Gradle this is done automatically via the separate task (see the `build.gradle` of this module).
+To run the test in Idea, you need to add the VM options manually in the "Edit configurations" tab. 
+Typically, the javac JAR can be found in the Gradle Caches directory in
```
Comment (armiol) [inline tools/errorprone-checks/src/test/java/README.md]:
```
`javac`
```
```
`caches`
```

File: tools/errorprone-checks/src/test/java/README.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+This tests require configuring `-Xbootclasspath...` option with the path to 
+the `com.google.errorprone.javac` JAR.
+
+In Gradle this is done automatically via the separate task (see the `build.gradle` of this module).
+To run the test in Idea, you need to add the VM options manually in the "Edit configurations" tab. 
+Typically, the javac JAR can be found in the Gradle Caches directory in 
+the `modules-2/files-2.1/com.google.errorprone/javac/` folder or its subfolders.
+After you acquired the path to the existing javac jar, add the following VM Option:
```
Comment (armiol) [inline tools/errorprone-checks/src/test/java/README.md]:
```
`javac`
```
`VM option`

File: tools/errorprone-checks/src/test/java/README.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+```
+-Xbootclasspath/p:`javacPath`
+```
+
+where the `javacPath` is the path to your javac JAR.
```
Comment (armiol) [inline tools/errorprone-checks/src/test/java/README.md]: Same as above.

File: tools/errorprone-checks/src/test/resources/io/spine/tools/check/methodresult/HandleMethodResultNegatives.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.check.vbuild;
```
Comment (armiol) [inline tools/errorprone-checks/src/test/resources/io/spine/tools/check/methodresult/HandleMethodResultNegatives.java]: It looks `... .given` to me.

File: tools/errorprone-checks/src/test/resources/io/spine/tools/check/methodresult/HandleMethodResultPositives.java [hunk 8]
```diff
@@ -18,33 +18,44 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-package io.spine.test.tools.check;
+package io.spine.tools.check.vbuild;
```
Comment (armiol) [inline tools/errorprone-checks/src/test/resources/io/spine/tools/check/methodresult/HandleMethodResultPositives.java]: Same as above.

**Comment by dmdashenkov** [comment]
@armiol, PTAL again.

File: tools/errorprone-checks/src/test/resources/io/spine/tools/check/methodresult/given/HandleMethodResultPositives.java [hunk 9]
```diff
@@ -18,33 +18,44 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-package io.spine.test.tools.check;
+package io.spine.tools.check.vbuild;
```
Comment (armiol) [inline tools/errorprone-checks/src/test/resources/io/spine/tools/check/methodresult/given/HandleMethodResultPositives.java]: Let's update the package.

File: tools/errorprone-checks/src/test/resources/io/spine/tools/check/methodresult/given/HandleMethodResultNegatives.java [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.check.vbuild;
```
Comment (armiol) [inline tools/errorprone-checks/src/test/resources/io/spine/tools/check/methodresult/given/HandleMethodResultNegatives.java]: And this one as well.

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

**Pull request merged**